framework/src/materials/MaterialPropertyStorage.C

```diff
@@ -350,6 +350,32 @@ MaterialPropertyStorage::swapBack(MaterialData & material_data, const Elem & ele
 }
 
 bool
+MaterialPropertyStorage::needInitialized(unsigned int n_qpoints, const Elem & elem, unsigned int side)
+{
+  bool need = false;
+
+  // Has not been intialized
+  if (props()[&elem][side].size() == 0 || propsOld()[&elem][side].size() == 0) need = true;
+  if (hasOlderProperties())
+    if (propsOlder()[&elem][side].size() == 0) need = true;
+
+  // The number of q_points has changed
+  if (!need)
```

Comment:
I understand that the material stateful properties are all initialized in ```FEProblem``` during initial setup. However, for ```DiracKernel```, it can not be done during initial setup (at that time, the system does not know any ```DiracKernel``` points) but has to do every time step if necessary. The code here will return true if the material property on this element is never initialized/allocated in ```MaterialPropertyStorage``` or if the number of points is changed on this element.